pkg/lib/template/template_loader.go

```diff
@@ -282,6 +283,42 @@ func (loader *Loader) Render(
 		"match": func(regex, text string) (bool, error) {
 			return regexp.Match(regex, []byte(text))
 		},
+		"replace_http_server": func(srcURL string) (string, error) {
+			// If no override provided, return original one
+			if loader.HTTPServer == "" {
+				return srcURL, nil
+			}
+			// Parse source URL or fail
+			parsedURL, err := url.Parse(srcURL)
+			if err != nil {
+				return "", err
+			}
+			// Parse override URL or fail
+			parsedOverrideURL, err := url.Parse(loader.HTTPServer)
+			if err != nil {
+				return "", err
+			}
+			// Go thru all provided parts and replace/add them
+			if parsedOverrideURL.Scheme != "" {
+				parsedURL.Scheme = parsedOverrideURL.Scheme
+			}
+			hostName := parsedOverrideURL.Hostname()
+			port := parsedOverrideURL.Port()
+			if hostName == "" && port != "" {
+				hostName = "localhost"
+			}
+			if hostName != "" {
+				parsedURL.Host = hostName
+			}		
+			if port != "" {
```

Comment:
Keep the overwrite ready to use, so the localhost fallback, put it into the topp.

Why are you mixing the overwrites here? Keep "port" and "hostname" separate. 

If the hostname happens to be "", use "localhost"